src/decrypter.js

```diff
@@ -290,75 +339,82 @@ decrypt = function(encrypted, key, initVector) {
   return decrypted;
 };
 
-AsyncStream = function() {
-  this.jobs = [];
-  this.delay = 1;
-  this.timeout_ = null;
-};
-AsyncStream.prototype = new videojs.Hls.Stream();
-AsyncStream.prototype.processJob_ = function() {
-  this.jobs.shift()();
-  if (this.jobs.length) {
-    this.timeout_ = setTimeout(this.processJob_.bind(this),
-                               this.delay);
-  } else {
+export class AsyncStream extends Stream {
+  constructor() {
+    super(Stream);
+    this.jobs = [];
+    this.delay = 1;
     this.timeout_ = null;
   }
-};
-AsyncStream.prototype.push = function(job) {
-  this.jobs.push(job);
-  if (!this.timeout_) {
-    this.timeout_ = setTimeout(this.processJob_.bind(this),
-                               this.delay);
+  processJob_() {
+    this.jobs.shift()();
+    if (this.jobs.length) {
+      this.timeout_ = setTimeout(
+        this.processJob_.bind(this),
+        this.delay
+      );
+    } else {
+      this.timeout_ = null;
+    }
   }
-};
+  push(job) {
+    this.jobs.push(job);
+    if (!this.timeout_) {
+      this.timeout_ = setTimeout(
+        this.processJob_.bind(this),
+        this.delay
+      );
+    }
+  }
+}
+
+// the maximum number of bytes to process at one time
+const decrypterStep = 4 * 8000;
 
-Decrypter = function(encrypted, key, initVector, done) {
-  var
-    step = Decrypter.STEP,
-    encrypted32 = new Int32Array(encrypted.buffer),
-    decrypted = new Uint8Array(encrypted.byteLength),
-    i = 0;
-  this.asyncStream_ = new AsyncStream();
-
-  // split up the encryption job and do the individual chunks asynchronously
-  this.asyncStream_.push(this.decryptChunk_(encrypted32.subarray(i, i + step),
-                                            key,
-                                            initVector,
-                                            decrypted));
-  for (i = step; i < encrypted32.length; i += step) {
-    initVector = new Uint32Array([
-      ntoh(encrypted32[i - 4]),
-      ntoh(encrypted32[i - 3]),
-      ntoh(encrypted32[i - 2]),
-      ntoh(encrypted32[i - 1])
-    ]);
+export class Decrypter extends Stream {
+  constructor(encrypted, key, initVector, done) {
+    super(Stream);
```

Comment:
Just noticing the Decrypter never invoked the Stream constructor. Seems like that inheritance is unnecessary.